core/trino-main/src/test/java/io/trino/sql/query/TestCheckConstraint.java

```diff
@@ -0,0 +1,177 @@
+                        return TPCH_NATION_SCHEMA;
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_not_boolean_expression"))) {
+                        return TPCH_NATION_SCHEMA;
+                    }
+                    throw new UnsupportedOperationException();
+                })
+                .withCheckConstraints(schemaTableName -> {
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation"))) {
+                        return ImmutableList.of("nationkey > 100");
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_multiple_column_constraint"))) {
+                        return ImmutableList.of("nationkey > 100 AND regionkey > 50");
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_invalid_function"))) {
+                        return ImmutableList.of("invalid_function(nationkey) > 100");
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_not_boolean_expression"))) {
+                        return ImmutableList.of("1 + 1");
+                    }
+                    throw new UnsupportedOperationException();
+                })
+                .withData(schemaTableName -> {
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation"))) {
+                        return TPCH_NATION_DATA;
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_multiple_column_constraint"))) {
+                        return TPCH_NATION_DATA;
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_invalid_function"))) {
+                        return TPCH_NATION_DATA;
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_not_boolean_expression"))) {
+                        return TPCH_NATION_DATA;
+                    }
+                    throw new UnsupportedOperationException();
+                })
+                .build();
+
+        runner.createCatalog(MOCK_CATALOG, mock, ImmutableMap.of());
+
+        assertions = new QueryAssertions(runner);
+    }
+
+    @AfterAll
+    public void teardown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
+    @Test
+    public void testInsert()
+    {
+        assertions.query("INSERT INTO mock.tiny.nation VALUES (101, 'POLAND', 0, 'No comment')")
+                .assertThat()
+                .skippingTypesCheck()
+                .matches("SELECT BIGINT '1'");
+
+        assertThatThrownBy(() -> assertions.query("INSERT INTO mock.tiny.nation VALUES (26, 'POLAND', 0, 'No comment')"))
+                .hasMessage("Cannot insert row that does not match to a check constraint: (\"nationkey\" > CAST(100 AS bigint))");
```

Comment:
Can we also have planner tests that assert actual plans?

@martint @kasiafi would it be a good idea here?